### PR TITLE
insights: default to fork:no archived:no for code insights queries

### PR DESCRIPTION
**Pull request by coury-clark**
Default code insights queries to using `fork:no` `archive:no`. This doesn't propagate to insights calculated on the frontend. This will preserve user inputted values if provided.

## Test plan

Example tests on the live preview resolver against a fork repo.

``` gql
{
  "input": {
    "series": [
      {
        "generatedFromCaptureGroups": false,
        "query": "Messages file:^protocol\\.md",
        "label": "abc"
      }
    ],
    "repositoryScope": {
      "repositories": [
        "github.com/sourcegraph/language-server-protocol"
      ]
    },
    "timeScope": {
      "stepInterval": {
        "unit": "MONTH",
        "value": 1
      }
    }
  }
}
```

``` json
{
  "data": {
    "searchInsightPreview": [
      {
        "points": [
          {
            "dateTime": "2021-11-11T00:00:00Z",
            "value": 0
          },
          {
            "dateTime": "2021-12-11T00:00:00Z",
            "value": 0
          },
          {
            "dateTime": "2022-01-11T00:00:00Z",
            "value": 0
          },
          {
            "dateTime": "2022-02-11T00:00:00Z",
            "value": 0
          },
          {
            "dateTime": "2022-03-11T00:00:00Z",
            "value": 0
          },
          {
            "dateTime": "2022-04-11T00:00:00Z",
            "value": 0
          },
          {
            "dateTime": "2022-05-11T00:00:00Z",
            "value": 0
          }
        ]
      }
    ]
  }
}
```

Example of one of the generated queries:
`archived:no fork:only file:^protocol\.md count:99999999 Messages repo:^github\.com/sourcegraph/language-server-protocol$@6f083d7cf03fb2a282b1f940d31c15b400ae7df6`

Changing the query to `Messages fork:yes file:^protocol\\.md`

``` json
{
  "data": {
    "searchInsightPreview": [
      {
        "points": [
          {
            "dateTime": "2021-11-11T00:00:00Z",
            "value": 5
          },
          {
            "dateTime": "2021-12-11T00:00:00Z",
            "value": 5
          },
          {
            "dateTime": "2022-01-11T00:00:00Z",
            "value": 5
          },
          {
            "dateTime": "2022-02-11T00:00:00Z",
            "value": 5
          },
          {
            "dateTime": "2022-03-11T00:00:00Z",
            "value": 5
          },
          {
            "dateTime": "2022-04-11T00:00:00Z",
            "value": 5
          },
          {
            "dateTime": "2022-05-11T00:00:00Z",
            "value": 5
          }
        ]
      }
    ]
  }
}
```

### CODE REVIEW

**Comment by coury-clark** [comment]
JIT testing:
CleanShot 2022-05-11 at 11 03 30@2x

Background calculated and filtered to a fork
CleanShot 2022-05-11 at 11 01 30@2x

File: enterprise/internal/insights/query/querybuilder/builder.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	modified := withCountAll(query)
+	modified, err := withDefaults(modified, codeInsightsQueryDefaults())
+	if err != nil {
+		return "", errors.Wrap(err, "WithDefaults")
+	}
+	modified = forRepoRevision(modified, repo, revision)
+
+	return modified, nil
+}
+
```
Comment (chwarwick) [inline enterprise/internal/insights/query/querybuilder/builder.go:82]: Any concerns with these two functions making the assumptions that no `repo` parameter is present in the search string?

**Review by chwarwick — APPROVED** [review]
Nice, I'm not sure for what yet, but I feel like having that builder is going to be helpful in the future.

File: enterprise/internal/insights/query/querybuilder/builder.go (hunk 1 above)
Comment (coury-clark) [reply enterprise/internal/insights/query/querybuilder/builder.go:82]: I don't think so - this is copy paste the same thing we've been doing all along. Ideally I want to refactor *all* of the query modifications to actually use the parser, but that's not blocking for this issue (I'll open new issues for these)

**Pull request merged**